internal/pkg/db/mongo/data.go

```diff
@@ -62,7 +71,8 @@ func (mc MongoClient) AddEvent(e correlation.Event) (string, error) {
 				return "", err
 			}
 
-			r.TimestampForAdd()
```

Comment:
Can we revert this change? This function handles the setting the times and we can leverage that in a central location and not duplicate the logic.

Reply:
No. This PR is to fix it. Read more #2310.
The creation of time is in a for loop, which causes the creation time to be different in milliseconds.

Reply:
You are correct. Ignore my previous comment.